src/test/java/de/tum/in/www1/artemis/ImprintResourceIntegrationTest.java

```diff
@@ -0,0 +1,241 @@
+package de.tum.in.www1.artemis;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.argThat;
+import static org.mockito.Mockito.mockStatic;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.mockito.MockedStatic;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import de.tum.in.www1.artemis.domain.Imprint;
+import de.tum.in.www1.artemis.domain.LegalDocumentLanguage;
+import net.minidev.json.JSONObject;
+
+@ExtendWith(MockitoExtension.class)
+class ImprintResourceIntegrationTest extends AbstractSpringIntegrationBambooBitbucketJiraTest {
+
+    private static final String TEST_PREFIX = "ir"; // only lower case is supported
+
+    @Test
+    void testGetImprint_unsupportedLanguageBadRequest() throws Exception {
+        request.get("/api/imprint?language=fr", HttpStatus.BAD_REQUEST, Imprint.class);
+    }
+
+    @Test
+    void testGetImprint_cannotReadFileInternalServerError() throws Exception {
+        try (MockedStatic<Files> mockedFiles = mockStatic(Files.class)) {
+            mockedFiles.when(() -> Files.exists(argThat(path -> path.toString().contains("_de")))).thenReturn(true);
+            mockedFiles.when(() -> Files.readString(argThat(path -> path.toString().contains("_de")))).thenThrow(new IOException());
+            request.get("/api/imprint?language=de", HttpStatus.INTERNAL_SERVER_ERROR, Imprint.class);
+        }
+
+    }
```

Comment:
Why doesn't this have to also use the admin test user?

Reply:
This endpoint is not protected as this endpoint returns the imprint if you want to view it. --> no special privileges required